macaroonbakery/bakery.py

```diff
@@ -0,0 +1,280 @@
+# Copyright 2016 Canonical Ltd.
+# Licensed under the LGPLv3, see LICENCE file for details.
+
+import base64
+import json
+import webbrowser
+
+import macaroons
+import requests
+
+
+def deserialize(json_macaroon):
+    '''Deserialize a JSON macaroon into a macaroon object.
+
+    @param the JSON macaroon to deserialize as a dict.
+    @return the deserialized macaroon object.
+    '''
+    return macaroons.deserialize(to_base64(json_macaroon))
+
+
+def serialize_macaroon_string(macaroon):
+    '''Serialize macaroon object to string.
+
+    @param macaroon object to be serialized.
+    @return a string serialization form of the macaroon.
+    '''
+    return base64.urlsafe_b64decode(
+        _add_padding(macaroon.serialize_json()))
+
+
+def to_base64(json_macaroon):
```

Comment:
ISTM that there's nothing macaroon-ish about this at all.
It's just a dict to JSON encoder,

Also, is it correct to use URL-safe encoding? I think the bakery
uses standard encoding.